src/ui/src/js/controllers/request_view.js

```diff
@@ -33,6 +33,33 @@ export default function requestViewController(
     request) {
   $scope.service = RequestService;
 
+ $scope.download = function() {
+      var filename = $scope.data.id;
+      var output;
+      var outputType = $scope.data.output_type;
+      if($scope.data.output === undefined || $scope.data.output == null){
+        window.alert("No output to download");
+        return;
+      } else if(outputType.toLowerCase() == "string"){
+        output = $scope.rawOutput;
+        filename = filename+".txt"
+      } else if(outputType.toLowerCase() == "html"){
+        output = $scope.htmlOutput;
+        filename = filename+".html"
+      } else if(outputType.toLowerCase() == "json"){
+        output = $scope.jsonOutput;
+        filename = filename+".json"
+      }
+      var element = document.createElement('a');
+      element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(output));
+      element.setAttribute('download', filename);
+      element.style.display = 'none';
+      document.body.appendChild(element);
+      element.click();
+      document.body.removeChild(element);
+      return;
```

Comment:
I'm not sure what's going on here, we'll have to chat about this one :smile:

Reply:
It creates a hyperlink for download then clicks hyperlink, then removes hyperlink. I will change the download button to a hyperlink of the code above since it does not need to create a hyperlink just to delete the hyperlink.